operator/src/main/java/org/keycloak/operator/KeycloakDeployment.java

```diff
@@ -0,0 +1,146 @@
+        return new DeploymentBuilder()
+                .withNewMetadata()
+                    .withName(Constants.NAME)
+                    .withNamespace(keycloak.getMetadata().getNamespace())
+                    .addToLabels(Constants.MANAGED_BY_LABEL, Constants.MANAGED_BY_VALUE)
+                    .addNewOwnerReference()
+                        .withApiVersion(Constants.CRDS_VERSION)
+                        .withKind(keycloak.getKind())
+                        .withName(keycloak.getMetadata().getName())
+                        .withUid(keycloak.getMetadata().getUid())
+                    .endOwnerReference()
+                .endMetadata()
+
+                .withNewSpec()
+
+                .withNewTemplate()
+                .withNewMetadata()
+                .addToLabels(Constants.DEFAULT_LABELS)
+                .endMetadata()
+                .withNewSpec()
+
+                .addNewInitContainer()
+                .withName("init-container")
+                .withImage(Constants.DEFAULT_KEYCLOAK_INIT_IMAGE)
+                .endInitContainer()
+
+                .addNewContainer()
+                .withName(Constants.NAME)
+                .withImage(Constants.DEFAULT_KEYCLOAK_IMAGE)
+                .addNewPort()
+                .withContainerPort(8443)
+                .withProtocol("TCP")
+                .endPort()
+                .endContainer()
+                .endSpec()
+
+                .endTemplate()
+                .withReplicas(keycloak.getSpec().getInstances())
+                .withNewSelector()
+                .addToMatchLabels(Constants.DEFAULT_LABELS)
+                .endSelector()
+
+                .endSpec()
+                .build();
+    }
+
+    public KeycloakStatus getNextStatus(KeycloakSpec desired, KeycloakStatus prev, Deployment current) {
+
+        var isReady = (current != null &&
+                current.getStatus() != null &&
+                current.getStatus().getReadyReplicas() != null);
+
+        if (prev == null) {
+            var newStatus = new KeycloakStatus();
+            newStatus.setState(READY);
+            newStatus.setMessage("Keycloak deployment started");
+            return newStatus;
+        }
+
+        switch (prev.getState()) {
+            case READY:
+                if (isReady) {
+                    return null;
+                } else {
+                    var newStatus = prev.clone();
+                    newStatus.setState(UNKNOWN);
+                    newStatus.setMessage("Keycloak deployment is NOT ready");
+                    return newStatus;
+                }
+            case ERROR:
+            case UNKNOWN:
+                if (isReady) {
+                    var newStatus = prev.clone();
+                    newStatus.setState(READY);
+                    newStatus.setMessage("Keycloak deployment is ready!");
+                    return newStatus;
+                } else {
+                    return null;
+                }
+        }
```

Comment:
This feels a bit cumbersome TBH. Couldn't we just always return the state (never return `null`) and just call `equals` in the controller to determine whether the state needs updating?

Reply:
If you prefer I can remove the handling of the status at all

Reply:
Yes, I think rather than having wrong logic (which is not apparent at the first sight), I'd rather remove it.

Reply:
@andreaTP Could you please mark this as resolved?